homeassistant/components/switch_as_x/__init__.py

```diff
@@ -100,6 +101,37 @@ async def async_registry_updated(
     return True
 
 
+async def async_migrate_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Migrate old entry."""
+    _LOGGER.debug(
+        "Migrating from version %s.%s", config_entry.version, config_entry.minor_version
+    )
+
+    if config_entry.version > 1:
+        # This means the user has downgraded from a future version
+        return False
+    if config_entry.version == 1:
+        options = {**config_entry.options}
+        if config_entry.minor_version < 2:
+            options[CONF_INVERT] = False
```

Comment:
This means that if someone has upgraded and configured a `switch_as_x` with `invert=True` and then downgrades it will change the option for that entry to `invert=False` hence changing how that entity works (even if the user then upgrade again).

It should probably only set this option if it's not already there?